src/witan/models/dem/ccm/core/projection_loop.clj

```diff
@@ -7,83 +7,135 @@
             [witan.datasets :as wds]
             [witan.models.dem.ccm.schemas :refer :all]))
 
-(defn get-last-yr-from-popn
-  "Takes in a dataset. Select the latest year and returns it.
-  Note: expect a :year column."
-  [popn]
-  (reduce max (i/$ :year popn)))
-
 (defworkflowfn keep-looping?
   {:witan/name :ccm-core/ccm-loop-pred
    :witan/version "1.0"
-   :witan/input-schema {:population PopulationSchema}
+   :witan/input-schema {:population PopulationSchema :loop-year s/Int}
    :witan/param-schema {:last-proj-year s/Int}
    :witan/output-schema {:loop-predicate s/Bool}
-   ;; :witan/predicate? true
    :witan/exported? true}
-  [{:keys [population]} {:keys [last-proj-year]}]
-  {:loop-predicate (< (get-last-yr-from-popn population) last-proj-year)})
+  [{:keys [loop-year]} {:keys [last-proj-year]}]
+  {:loop-predicate (< loop-year last-proj-year)})
 
-(defworkflowfn select-starting-popn
-  "Takes in a dataset of popn estimates.
-   Returns a dataset where the last year's population from the input data
-   is appended as the starting population for the next year's projection"
+(defworkflowfn prepare-inputs
   {:witan/name :ccm-core/get-starting-popn
    :witan/version "1.0"
    :witan/input-schema {:population PopulationSchema}
    :witan/param-schema {:_ nil}
-   :witan/output-schema {:population PopulationSchema}}
+   :witan/output-schema {:loop-year s/Int :latest-yr-popn PopulationSchema}}
   [{:keys [population]} _]
-  (let [latest-yr (get-last-yr-from-popn population)
-        last-yr-data (i/query-dataset population {:year latest-yr})
-        update-yr (i/replace-column :year (i/$map inc :year last-yr-data) last-yr-data)]
-    {:population (ds/join-rows population update-yr)}))
+  (let [last-yr (reduce max (i/$ :year population))
+        last-yr-popn (i/query-dataset population {:year last-yr})]
+    {:loop-year last-yr :latest-yr-popn last-yr-popn}))
 
-(defworkflowfn age-on
+(defworkflowfn select-starting-popn
   "Takes in a dataset of popn estimates.
-   Returns a dataset where the last year's population is aged on 1 year.
+   Returns a dataset of the starting population for the next year's projection."
+  {:witan/name :ccm-core/get-starting-popn
+   :witan/version "1.0"
+   :witan/input-schema {:latest-yr-popn PopulationSchema :population PopulationSchema
+                        :loop-year s/Int}
+   :witan/param-schema {:_ nil}
+   :witan/output-schema {:latest-yr-popn PopulationSchema :loop-year s/Int}}
+  [{:keys [latest-yr-popn population loop-year]} _]
+  (let [update-yr (i/replace-column :year (i/$map inc :year latest-yr-popn) latest-yr-popn)]
+    {:latest-yr-popn update-yr :loop-year (inc loop-year)}))
+
+(defworkflowfn age-on
+  "Takes in a dataset with the starting-population.
+   Returns a dataset where the population is aged on 1 year.
    91 year olds are added to the 90 year olds in the aged-on popn."
```

Comment:
Could you change the comment to: Last year's 89 year olds are aged on and added to this year's 90+ age group (represented in code as just age 90)